hyperspy/io_plugins/emd.py

```diff
@@ -0,0 +1,392 @@
+# Writing features
+writes = True
+EMD_VERSION = '0.2'
+# ----------------------
+
+
+class EMD(object):
+
+    """Class for storing electron microscopy datasets.
+
+    The :class:`~.EMD` class can hold an arbitrary amount of datasets in the `signals` dictionary.
+    These are saved as HyperSpy :class:`~hyperspy.signal.Signal` instances. Global metadata
+    are saved in four dictionaries (`user`, `microscope`, `sample`, `comments`). To print
+    relevant information about the EMD instance use the :func:`~.print_info` function. EMD
+    instances can be loaded from and saved to emd-files, an hdf5 standard developed at Lawrence
+    Berkeley National Lab (http://emdatasets.lbl.gov/).
+
+    Attributes
+    ----------
+    signals: dictionary
+        Dictionary which contains all datasets as :class:`~hyperspy.signal.Signal` instances.
+    user: dictionary
+        Dictionary which contains user related metadata.
+    microscope: dictionary
+        Dictionary which contains microscope related metadata.
+    sample: dictionary
+        Dictionary which contains sample related metadata.
+    comments: dictionary
+        Dictionary which contains additional commentary metadata.
+
+    """
+
+    _log = logging.getLogger(__name__)
+
+    def __init__(self, signals=None, user=None, microscope=None, sample=None, comments=None):
+        self._log.debug('Calling __init__')
+        # Create dictionaries if not present:
+        if signals is None:
+            signals = {}
+        if user is None:
+            user = {}
+        if microscope is None:
+            microscope = {}
+        if sample is None:
+            sample = {}
+        if comments is None:
+            comments = {}
+        # Make sure some default keys are present in user:
+        for key in ['name', 'institution', 'department', 'email']:
+            if key not in user:
+                user[key] = ''
+        self.user = user
+        # Make sure some default keys are present in microscope:
+        for key in ['name', 'voltage']:
+            if key not in microscope:
+                microscope[key] = ''
+        self.microscope = microscope
+        # Make sure some default keys are present in sample:
+        for key in ['material', 'preparation']:
+            if key not in sample:
+                sample[key] = ''
+        self.sample = sample
+        # Add comments:
+        self.comments = comments
+        # Make sure the signals are added properly to signals:
+        self.signals = {}
+        for name, signal in signals.iteritems():
+            self.add_signal(signal, name)
+
+    def __getitem__(self, key):
+        # This is for accessing the raw data easily. For the signals use emd.signals[key]!
+        return self.signals[key].data
+
+    def _write_signal_to_group(self, signal_group, signal):
+        self._log.debug('Calling _write_signal_to_group')
+        # Save data:
+        dataset = signal_group.create_group(signal.metadata.General.title)

```

Comment:
I changed it everywhere you suggested. I think this will only be a problem if you save a signal to an already existing .emd-file and it will then still overwrite the data in the `dataset` group but other stuff saved there is kept intact. I'm not sure myself which behaviour would be preferable, `create_group` would ensure you start with a blank slate for the signal (if you save a completely different signal under an existing name, metadata which have nothing to do with the new signal could still be there...).